lib/ClangImporter/ImportDecl.cpp

```diff
@@ -5636,6 +5660,65 @@ Decl *SwiftDeclConverter::importGlobalAsInitializer(
   return result;
 }
 
+Decl *SwiftDeclConverter::importCXXMethodDecl(
+    const clang::CXXMethodDecl *decl, DeclName name, DeclContext *dc,
+    Optional<ImportedName> correctSwiftName) {
+  if (decl->getAccess() != clang::AS_public) return nullptr;
```

Comment:
This is an interesting suggestion (and the general one of sharing more code). There is lots of duplication already in this area of the code. It is hard to tell quite for sure how necessary it is. For example, all three languages so far have different notions of currying the self parameter that are structurally very similar. As this is a ways from being complete (I expect more details about const refs& etc). One benefit of keeping it separate for now is that there is less of a chance of breaking the other importers. One other downside of unifying with the c version is that it may make it harder to see and refactor out equivalence between c++ methods and other types of functions like swift and objective-c.

For reference, the general c++ parts so far are constness and the isStatic.

Reply:
My worry is that we'll need to make a change to how C functions import and we'll forget to do the same thing for C++ methods. But I'll leave it up to you (unless John has a strong opinion).

Reply:
Sounds reasonable. I'll work to put together a separate PR without the CXX abstractions, and we can compare and contrast / come back to this if it proves necessary.

Reply:
To be clear, I think we want all the changes around `AbstractionPattern`; it would just be nice if the code path for C++ methods was 90% the same as for a C function.

Reply:
Alright, I made that change. Hopefully Jordan is happy now.

Reply:
Yeah, I like how simple it became. Thanks, @pschuh!